### PR TITLE
New serverless pattern - Kinesis - EventBridge Pipes (filter + enrichment) - SFN

### PR DESCRIPTION
**Pull request by aphilippartd**
*Issue #, if available:* https://github.com/aws-samples/serverless-patterns/issues/2234

*Description of changes:* Create an EventBridge Pipe to connect a Kinesis Data Stream and a Step Functions Workflow while filtering and enriching messages

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kinesis-eventbridge-pipes-filter-enrichment-sfn/images/2.png
Comment (parikhudit) [inline kinesis-eventbridge-pipes-filter-enrichment-sfn/images/2.png:1]: Image contains PII. Please delete it and update one that masks PII like account numbers.

File: kinesis-eventbridge-pipes-filter-enrichment-sfn/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+This pattern shows how to build a pipeline that filters and enriches messages comming in an Amazon Kinesis Stream to an AWS Step Function Workflow. It leverages EventBridge Pipes to connect the source stream and the target workflow. On top of just connecting a source and a target, EventBridge Pipes allow to filter and enrich messages before they reach the target.
+
+## Testing
+
+Follow the steps below to test out the pattern:
+1. Grab the Amazon Kinesis Stream name from the SAM deployment outputs
+2. Run the following commands from your terminal
+   ```curl
+    aws kinesis put-record --stream-name REPLACE_YOUR_STREAM_NAME --data '{ "message": "Hello World", "category": "PASS_FILTER" }' --partition-key MY_PARTITION_KEY
```
Comment (parikhudit) [inline kinesis-eventbridge-pipes-filter-enrichment-sfn/README.md:49]: AWS CLI v2 required "--cli-binary-format raw-in-base64-out". Please test it and update README accordingly.

File: kinesis-eventbridge-pipes-filter-enrichment-sfn/images/1.png
[the remark this answers is not in the document]
Comment (aphilippartd) [reply kinesis-eventbridge-pipes-filter-enrichment-sfn/images/1.png:1]: Masked

File: kinesis-eventbridge-pipes-filter-enrichment-sfn/images/2.png
Comment (aphilippartd) [reply kinesis-eventbridge-pipes-filter-enrichment-sfn/images/2.png:1]: Masked

File: kinesis-eventbridge-pipes-filter-enrichment-sfn/README.md (hunk 1 above)
Comment (aphilippartd) [reply kinesis-eventbridge-pipes-filter-enrichment-sfn/README.md]: Done

**Comment by parikhudit** [comment]
Thanks for the contribution @aphilippartd. Our DA team will merge the changes soon.

**Pull request merged**